packages/container/libraries/strongly-typed/src/container.ts

```diff
@@ -0,0 +1,210 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable @typescript-eslint/no-unnecessary-type-parameters */
+import { Container, type interfaces } from 'inversify';
+
+type IfAny<T, TYes, TNo> = 0 extends 1 & T ? TYes : TNo;
+
+type BindingMapProperty = string | symbol;
+type BindingMap = Record<BindingMapProperty, any>;
```

Comment:
Should we protect from async type mappings such as:

```ts
interface MyBindingMap {
  foo: Promise<Foo>;
}
```

I think we should add this to the docs or do some type mapping such as:

```ts
AwaitedBindingMap<TBindingMap extends BindingMap> = {
  [TKey in keyof TBindingMap]: Awaited<TBindingMap[TKey]>
}
```
This way we could receive a `BindingMap` and internally apply `AwaitedBindingMap<BindingMap>` to the `TypeContainer`.

Reply:
I'm not sure I follow? I thought we allowed promises, and just enforced that they had to be accessed with `getAsync()`? Basically this test case: https://github.com/inversify/monorepo/pull/59/files#diff-9892fb8135f8ba0c3d8719dc0d16884330ab5a7651ac09aa65ffe97712f6784eR121-R128

I'm also updated the `README` to talk about `Promise`s.

Reply:
You're right, we agreed on that. What should we do about the following update in your test:

```ts
it('gets a promise', async () => {
  // @ts-expect-error :: can't call get() to get Promise
  expect(() => container.get('asyncNumber')).toThrow(
    'it has asynchronous dependencies',
  );
  const n: Promise<number> = container.getAsync('asyncNumber');
  expect(await n).toBe(1);
});
```
It complains with the following error:

```
Type 'Promise<Promise<number>>' is not assignable to type 'Promise<number>'.
  Type 'Promise<number>' is not assignable to type 'number'.
```

But we do expect a `Promise<number>` instead of a `Promise<Promise<number>>`.